sunpy/image/slit.py

```diff
@@ -0,0 +1,114 @@
+import sunpy.map
+
+import numpy as np
+
+import astropy.units as u
+
+
+def slit(mcube_in, range_a, range_b):
+    """
+    Returns an array with intensity along the slit on the y axis and time 
+    along x.
+    
+    Parameters    
+    ----------
+    mcube_in : `sunpy.map.MapCube`
+        A mapcube of the images you want to perform the slit analysis on. 
+        Usually with x and y as space, and z as time.
+    range_a : `astropy.units.Quantity` 
+        A list of two `astropy.unit.Quantity` objects representing x1 and x2, 
+        start and end of slit in x.
+    range_b : `astropy.units.Quantity`
+        A list of two `astropy.unit.Quantity` objects representing x2 and y2, 
+        start and end of slit in y.
+
+    Returns
+    -------
+    out : `numpy.ndarray`
+        A numpy array with intenisty in y and time in x.
+
+    """
+    
+    # check the attributes of the coordinates
+    if ((isinstance(range_a, u.Quantity) and isinstance(range_b, u.Quantity)) or
+        (hasattr(range_a, 'unit') and hasattr(range_b, 'unit'))):
+    
+        if (range_a.unit.is_equivalent(mcube_in[0].units.x) and
+            range_b.unit.is_equivalent(mcube_in[0].units.y)):
+            
+            # convert the world to pixel
+            init_map = sunpy.map.Map(mcube_in[0])
+            c_x1, c_y1 = init_map.data_to_pixel(range_a[0], range_b[0])
+            c_x2, c_y2 = init_map.data_to_pixel(range_a[1], range_b[1])            
+                        
+        elif range_a.unit.is_equivalent(u.pixel) and range_b.unit.is_equivalent(u.pixel):
+            c_x1, c_y2 = range_a[0], range_b[0]
```

Comment:
oh, I see now you are using `.value` below, which is fine, you could still reorder the unpacking as above, to not have to index things if you want.